### PR TITLE
Reconciliation added to KafkaAvailability logging (#4647)

### PR DESCRIPTION
**Pull request by LiamClarkeNZ**
### Type of change

Enhancement / new feature

### Description

* Added Reconciliation instance to logging in KafkaAvailability
* Added Reconciliation instance to one logging call missing it in KafkaRoller
* Added shellcheck to prerequisites_check.sh

### Checklist

- N/A Write tests
- [X] Make sure all tests pass
- N/A Update documentation
- N/A Check RBAC rights for Kubernetes / OpenShift roles
- [X] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [X] Reference relevant issue(s) and close them after merging
- N/A? Update CHANGELOG.md
- N/A Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAvailability.java [hunk 1]
```diff
@@ -55,17 +61,17 @@
...
 
     private Future<Boolean> canRollBroker(Future<Collection<TopicDescription>> descriptions, int podId) {
         Future<Set<TopicDescription>> topicsOnGivenBroker = descriptions
                 .compose(topicDescriptions -> {
-                    log.debug("Got {} topic descriptions", topicDescriptions.size());
+                    log.debug("{}: Got {} topic descriptions", reconciliation, topicDescriptions.size());
                     return Future.succeededFuture(groupTopicsByBroker(topicDescriptions, podId));
                 }).recover(error -> {
-                    log.warn(error);
+                    log.warn(new ParameterizedMessage("{}: failed to get topic descriptions", reconciliation), error);
```
Comment (tombentley) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAvailability.java:74]: Why not just `log.warn("{}: failed to get topic descriptions", reconciliation, error)`? (A "spare" exception argument, for which there is no `{}`, still gets dumped to the log).

Comment (LiamClarkeNZ) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAvailability.java:74]: Oh sweet, that's way better - I was wondering why my IDE hadn't highlighted it as a warning, and thought it was just an IDE issue.

Comment (LiamClarkeNZ) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAvailability.java]: Changes made, thanks @tombentley

**Comment by LiamClarkeNZ** [comment]
> Commenter does not have sufficient privileges for PR 4709 in repo strimzi/strimzi-kafka-operator

:(

**Comment by Frawless** [comment]
@LiamClarkeNZ build pipeline si triggered automatically with every push. Do you want to run some other with system tests?

**Comment by LiamClarkeNZ** [comment]
> @LiamClarkeNZ build pipeline si triggered automatically with every push. Do you want to run some other with system tests?

Kia ora @Frawless it looked like the last build step failed due to a slightly flaky test, just wanted to rerun the build step of the pipeline, thought I'd see if the Azure integration worked. I've just pushed a slight change to the commit to trigger the build again.

**Comment by Frawless** [comment]
@LiamClarkeNZ could you double-check if you can rerun it manually via gh ui or on azure ui? In azure it's here 
image for gh it's in check somewhere. To run pipelines by writing comment you need write rights into repository unfortunately :/

**Comment by LiamClarkeNZ** [comment]
> @LiamClarkeNZ could you double-check if you can rerun it manually via gh ui or on azure ui? 

I can't currently, I've created an account on Azure Pipelines, but no rerun buttons, can't find similar in GH UI either.

Screenshot_20210413_163719

**Review by tombentley — APPROVED** [review]

**Pull request merged**